companion/lib/Surface/Handler.ts

```diff
@@ -582,6 +583,25 @@ export class SurfaceHandler extends EventEmitter<SurfaceHandlerEvents> {
 		}
 	}
 
+	#onDeviceSwipePage(forward: boolean): void {
+		if (this.#isSurfaceLocked || !this.panel) return
+
+		const pageNumber = this.#pageStore.getPageNumber(this.#currentPageId)
+		if (!pageNumber) return
+
+		if (this.#surfaces.deviceAllowsSwipeToChangePage(this.surfaceId, true)) {
```

Comment:
3) then this check should be entirely handled inside the stream deck implementation, as being able to disable this is a choice made by the stream deck implementation